taigit/src/libraries/GitHub/src/avgComments.ts

```diff
@@ -0,0 +1,58 @@
+import axios from 'axios';
+
+export async function
+getNumComments(owner : string, repo: string){
+    try{
+        let issue_comment = await axios.get("https://api.github.com/repos/" + owner +
+            "/" + repo + "/issues?state=closed");
+        let lengthOfClosedPullRequests = issue_comment.data.length
+        let issueNumberArray = Array()
+        console.log("Total No. Of Closed Pull Requests: "+lengthOfClosedPullRequests)
+        let totalComments = 0
+        issue_comment.data.forEach(function(req : {url:string} ) {
+            let closedIssueNumberArray = req.url.split('/');
+            let issue_number: string = closedIssueNumberArray[closedIssueNumberArray.length - 1];
+            issueNumberArray.push(issue_number)
+
+        });
+        for(let issue_number of issueNumberArray){
+            let commentsPerIssue = await getNumberCommentsPerPullRequest(owner,repo,Number(issue_number))
+            totalComments = totalComments + Number(commentsPerIssue)
+            console.log("Comments for Issue "+issue_number+" = " + Number(commentsPerIssue))}
+
+        console.log("Total Number Of Comments on PR: "+totalComments)
+        let averageNoOfComments = (totalComments/lengthOfClosedPullRequests).toFixed(2)
+        console.log("Average Number Of Comments: " +averageNoOfComments)
+        return averageNoOfComments;
+
+    }
+    catch (error) {
+        console.log(error);
```

Comment:
The error console logs should stay though as it'll help debug in case things go wrong.

Reply:
I know. Commented out console.log used for testing.